src/main/java/li/cil/sedna/riscv/R5CPUTemplate.java

```diff
@@ -476,15 +499,19 @@ private boolean csrrscx(final int rd, final int rs1, final int csr, final long m
 
         return false;
     }
+    private boolean isReadonlyCSR(final int csr) {
+        // Topmost bits, i.e. csr[11:8], encode access rights for CSR by convention. Of these, the top-most two bits,
+        // csr[11:10], encode read-only state, where 0b11: read-only, 0b00..0b10: read-write.
+        boolean readonly = ((csr & 0b1100_0000_0000) == 0b1100_0000_0000);
+        // There are also these special cases
+        readonly |= (csr >= 0xC00 && csr <= 0xC1F) || (csr >= 0xC80 && csr <= 0xC9F);
+        return readonly;
+    }
 
     private void checkCSR(final int csr, final boolean throwIfReadonly) throws R5IllegalInstructionException {
-        if (throwIfReadonly && ((csr >= 0xC00 && csr <= 0xC1F) || (csr >= 0xC80 && csr <= 0xC9F)))
+        if (throwIfReadonly && isReadonlyCSR(csr))
             throw new R5IllegalInstructionException();
 
-        // Topmost bits, i.e. csr[11:8], encode access rights for CSR by convention. Of these, the top-most two bits,
-        // csr[11:10], encode read-only state, where 0b11: read-only, 0b00..0b10: read-write.
-        if (throwIfReadonly && ((csr & 0b1100_0000_0000) == 0b1100_0000_0000))
-            throw new R5IllegalInstructionException();
         // The two following bits, csr[9:8], encode the lowest privilege level that can access the CSR.
```

Comment:
Maybe slightly reword the comment, now that the leading comment that led into this one has been moved to a different method? E.g. just drop the first 4 words.